src/event-helper.js

```diff
@@ -108,44 +108,52 @@ export function isLoaded(element) {
 export function loadPromise(element, opt_timeout) {
   let unlistenLoad;
   let unlistenError;
-  const loadingPromise = new Promise((resolve, reject) => {
-    if (isLoaded(element)) {
-      resolve(element);
+  if (isLoaded(element)) {
```

Comment:
Thoughts on using deferred objects to eliminate more closures? Not suggesting we use the deferred pattern instead of directly constructing promises everywhere, but just a few places it can be helpful like here where it can remove the closure around element

```
let loadingDeferred = new Deferred();
unlistenLoad = listenOnce(element, 'load', loadingDeferred.resolve);
unlistenError = listenOnce(element, 'error', loadingDeferred.reject);
let loadingPromise = loadingDeferred.promise.then(getTarget, failedToLoad);
```